newrelic-php-agent/templates/deployment.yaml

```diff
@@ -0,0 +1,110 @@
+{{- if and .Values.newrelic.deployment.enabled (not .Values.newrelic.daemonset.enabled) }}
+{{- $root := . -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "newrelic-php-agent.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "newrelic-php-agent.name" . }}
+    helm.sh/chart: {{ include "newrelic-php-agent.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.newrelic.deployment.replicaCount }}
+{{- with .Values.newrelic.strategy }}
+  strategy:
+{{ toYaml . | indent 4 }}
+{{- end }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "newrelic-php-agent.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "newrelic-php-agent.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: "{{ include "newrelic-php-agent.fullname" . }}"
+        image: "{{ .Values.newrelic.image.repository }}:{{ .Values.newrelic.image.tag }}"
+        imagePullPolicy: {{ .Values.newrelic.image.pullPolicy }}
+{{- if isAbs $.Values.newrelic.address }}
+{{- with .Values.newrelic.image.command }}
+        command:
+{{ tpl (toYaml .) $root | indent 10 }}
+{{- end }}
+{{- else }}
+        command:
+          - --address={{ $.Values.newrelic.address }} --watchdog-foreground
```

Comment:
@cw-niinuma 
I think this is illegal as a `deployment`. Did it work?

Reply:
I suggest that it would be nice to use `args`.

```yaml
{{- with .Values.newrelic.image.command }}
command:
{{ tpl (toYaml .) $root | indent 10 }}
{{- end }} 
args:
- --address
- '{{ .Values.newrelic.address }}'
- --watchdog-foreground
{{- with .Values.newrelic.extraArgs }}
{{ tpl (toYaml .) $root | indent 10 }}
{{- end }} 
```